content/blog/2025-05-13-brilooped/index.md

```diff
@@ -0,0 +1,443 @@
+- `continue`: Takes one constant argument and exits `value` loops. This continues execution at the following instruction.
+
+For example the following program will print 1 and 2 forever:
+
+```
+cond: bool = const true;
+one: int = const 1;
+two: int = const 2;
+while cond {
+    print one;
+    while cond {
+        print two;
+        continue 1;
+    }
+}
+```
+
+Whereas this program will print 1 followed by infinite twos:
+
+```
+cond: bool = const true;
+one: int = const 1;
+two: int = const 2;
+while cond {
+    print one;
+    while cond {
+        print two;
+        continue 0;
+    }
+}
+```
+
+## Interpreter Changes
+
+### New Language Constructs
+
+We expanded Bril's control flow capabilities by introducing five new constructs to the interpreter type system:
+
+- `while` - For creating loops
+- `block` - For grouping instructions
+- `if` - For conditional execution
+- `break` - For exiting loops
+- `continue` - For skipping to the next loop iteration
+
+### Implementation Challenges
+
+The implementation complexity varied across these new constructs. The `break` and `continue` instructions were relatively straightforward as they act as terminals in the evaluation pipeline - they simply return an action to be handled by their parent control structure.
+
+However, the other three constructs (`while`, `block`, and `if`) demanded more sophisticated handling:
+
+```typescript
+case "if": {
+    if (cond) {
+        return evalBlock(children[0], state);
+    }
+    else {
+        return evalBlock(children[1], state);
+    }
+}
+
+case "block": {
+    const result = evalBlock(children[0], state);
+    if (result.action == "break") {
+        if result.level > 0 {
+            return { action: "break", level: result.level - 1 };
+        }
+        else {
+            return NEXT;
+        }
+    } else if (result.action === "continue") {
+        if (result.level > 0) {
+            return { action: "continue", level: result.level - 1 };
+        } else {
+            return NEXT;
+        }
+    } else {
+        return result;
+    }
+}
+
+case "while": {
+    while true {
+        if (!cond) {
+            return NEXT:
+        }
+        const result = evalBlock(children[0], state);
+        if (result.action == "break") {
+            if (result.level > 0) {
+                return { action: "break", level: result.level - 1 };
+            } else {
+                return NEXT;
+            }
+        } else if result.action == "continue" {
+            if (result.level > 0) {
+                return { action: "continue", level: result.level - 1 };
+            } else {
+                continue;
+            }
+        } else if (
+            result.action === "end" ||
+            result.action === "speculate" ||
+            result.action === "commit" ||
+            result.action === "abort"
+        ) {
+            return result;
+        }
+    }
+}
+
+```
```

Comment:
Can I encourage you to avoid the temptation to include big blocks of the implementation code here in the blog post? You can always include a link to the GitHub blob page for people who are really interested! But from the reader's perspective, just one or two sentences that briefly captures the _idea_ of the code can be way more valuable than the full code itself.